### ISSUE TITLE
Event listener for notification not working properly

### ISSUE THREAD

**Issue by m-dary-nur** [OneSignal/react-onesignal#25]

**Comment by pedro-lb** [comment]
Hey there @m-dary-nur , can you provide more details?

**Comment by thedeephakani** [comment]
same for me 
when I try to add an event using the below code it shows me an error that callback is missing
const events = [
                {
                    listener: "once",
                    event: "subscriptionChange",
                    callback: async (isSubscribed: boolean) => {
                        if (true === isSubscribed) {
                            console.log("The user subscription state is now:", isSubscribed);
                        }
                    },
                },
                {
                    event: "notificationDisplay",
                    callback: (event: any) => {
                        console.warn("OneSignal notification displayed:", event);
                    },
                },
                {
                    event: "notificationDismiss",
                    callback: (event: any) => {
                        console.warn("OneSignal notification dismissed:", event);
                    },
                },
            ];

Types of property 'callback' are incompatible.
        Property 'callback' is missing in type '(isSubscribed: boolean) => Promise<void>' but required in type 'IOneSignalEventCallback'
        
     after that I try to write event like below 
     
     const events = [
                {
                    listener: "once",
                    event: "subscriptionChange",
                    callback: {
                        callback: (isSubscribed: boolean) => {
                            if (true === isSubscribed) {
                                console.log("The user subscription state is now:", isSubscribed);
                            }
                        },
                    },
                },
                {
                    event: "notificationDisplay",
                    callback: {
                        callback: (event: any) => {
                            console.warn("OneSignal notification displayed:", event);
                        },
                    },
                },
                {
                    event: "notificationDismiss",
                    callback: {
                        callback: (event: any) => {
                            console.warn("OneSignal notification dismissed:", event);
                        },
                    },
                },
            ];
            
            
            so now it does not display any error but the calls back function is not working

**Comment by emzoumpo** [comment]
@pedro-lb This is solved in #23 but it's not released yet as a new version. @ 
m-dary-nur @ 
thedeephakani The implementation above is correct with the types of the current version, but the types are wrong, and that's why it's not working
If you ts-ignore the error and use
```javascript
{
                event: "notificationDismiss",
                    callback: (event: any) => {
                        console.warn("OneSignal notification dismissed:", event);
                    },
            },```
it should work

**Comment by pedro-lb** [comment]
Hey guys!

@emzoumpo @thedeephakani @m-dary-nur 

Please try our new version, details here: https://github.com/OneSignal/react-onesignal/issues/44

You'll need to change the initialization logic to set it up, but it should work much better ❤️
